### ISSUE TITLE
VS Code doesn't act correctly when re-using a merged PR's branch (was #179978)

### ISSUE THREAD

**Issue by brianjmurrell** [microsoft/vscode#185578]
This is being opened as a continuation of #179978.  (Less than even,) 2 weeks is not a lot of time to allow somebody to respond to an issue.  People can be quite busy and responding to issues might not be their highest priority task and responding might take some effort.  Closing an issue that is unable to be reopened after such a short period of time is just nasty.  Please refer to #179978 for details related to this issue.  I will not repeat that info here.

This issue can be seen simply by creating a new branch (`git checkout -b …`) via the CLI.  Once the new branch is created/checked out, VS Code is inconsistent in it's display of the branch and actions.  Witness:

![image](https://github.com/microsoft/vscode/assets/559317/0a050b7e-b23f-422d-a3f7-549eefe45d45)

As you can see, VS Code sees that it is on the new branch, but also displays that the branch is 1 commit behind and 91 commits ahead of the branch it was created from.  But after a `git checkout -b foobar/test-with-el8.7 origin/some-other-branch`, why would I want to push that to the branch that it was created from?.  99.9% of the time, any push of a newly created branch created from another branch will be to the new branch, not the one it was created from.

Very simply to reproduce.  Just open a terminal window and run a `git checkout -b ...` and see if you think how VS Code is displaying and handling the result is what you would expect.

After the above, I would expect VS Code to act as if I had created the branch using the VS Code UI's _Create new branch from_ facility.  It should show that a branch is new and pushable and once that is done should do the regular process of prompting to create a PR, etc.

Tagging @lszomoru per #179978.

**Comment by gjsjohnmurray** [comment]
@brianjmurrell I am not seeing the original issue as locked, so as author you should have been able to reopen it and add a comment.

**Comment by brianjmurrell** [comment]
@gjsjohnmurray Hrm, yes, you are correct.  I must have mistaken the greyed-out-Comment-button-before-you-type-into-the-comment-box as the issue could not be reopened.  I have added a comment there.

Ultimately I think this and #179978 are probably two different symptoms of the same problem -- VS Code getting out of sync with repo state changes happening outside of VS Code.

**Comment by brianjmurrell** [comment]
Actually @gjsjohnmurray you are (partially) incorrect.  I was able to add a comment to #179978 but it remains **Closed**.

**Comment by lszomoru** [comment]
@brianjmurrell, I went through the steps and here are my observations:
1) VS Code operates based on the information that is available locally in the `.git` folder.
2) After the topic branch has been deleted, as part of merging the PR you will have to run `git fetch --prune` so that your local branch is updated with the information that the remote tracking branch has been deleted.
 3) Even after running `git fetch --prune` VS Code is not picking up the fact that the branch does not have a remote so there is certainly a bug there that I would investigate and fix.

**Comment by brianjmurrell** [comment]
@lszomoru Indeed.  I realized that I had to do item 2 and actually always update my local repo.  In fact I wish there were a way to have VS code add `--prune` to it's `git fetch`.

It is the bug in item 3 that you acknowledge that I am reporting here.

**Comment by lszomoru** [comment]
> In fact I wish there were a way to have VS code add --prune to it's git fetch.

There is a "Git: Fetch (prune)" command in the command palette and in the `...` menu of the Source Control view that you can use. Also until the issue is fixed you can also run `git branch --unset-upstream` to unset the upstream of the local branch.
